diff --git a/Kahi_dspace_works/kahi_dspace_works/Kahi_dspace_works.py b/Kahi_dspace_works/kahi_dspace_works/Kahi_dspace_works.py
index b80f54a..7c356b6 100644
--- a/Kahi_dspace_works/kahi_dspace_works/Kahi_dspace_works.py
+++ b/Kahi_dspace_works/kahi_dspace_works/Kahi_dspace_works.py
@@ -147,6 +147,10 @@ class Kahi_dspace_works(KahiBase):
                 f"INFO: Processing repository {repository['institution_id']} collection {repository['collection_name']} with url {repository['repository_url']}")
             affiliation = process_affiliation(
                 repository["institution_id"], self.db)
+            if affiliation is None:
+                print(
+                    f"WARNING: Affiliation not found for repository {repository['institution_id']}. Skipping repository {repository['collection_name']}.")
+                continue
             base_url = repository["repository_url"]
             dspace_collection = dsapce_db[repository["collection_name"]]
             self.process_repository(affiliation, base_url, dspace_collection)
diff --git a/Kahi_dspace_works/kahi_dspace_works/utils.py b/Kahi_dspace_works/kahi_dspace_works/utils.py
index 4be0360..533a15c 100644
--- a/Kahi_dspace_works/kahi_dspace_works/utils.py
+++ b/Kahi_dspace_works/kahi_dspace_works/utils.py
@@ -67,10 +67,12 @@ def process_affiliation(ror_id, db):
 
     Returns
     -------
-    dict
-        affiliation processed.
+    dict | None
+        affiliation processed or None if affiliation is not found in database.
     """
     aff_rec = db["affiliations"].find_one({"external_ids.id": ror_id})
+    if aff_rec is None:
+        return None
     aff = {}
     aff["id"] = aff_rec["_id"]
     aff["name"] = next((i["name"]
